src/Analyzers/CSharp/Tests/RemoveRedundantElseStatement/RemoveRedundantElseStatementTests.cs

```diff
@@ -0,0 +1,913 @@
+                """);
+        }
+
+        [Fact]
+        public async Task TestRedundantElseFix_GlobalStatement()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode = """
+                    using System;
+                    
+                    if (false)
+                    {
+                        return;
+                    }
+                    [|else|]
+                    {
+                        Console.WriteLine("Success");
+                    }
+                    """,
+                FixedCode = """
+                    using System;
+                    
+                    if (false)
+                    {
+                        return;
+                    }
+
+                    Console.WriteLine("Success");
+
+                    """,
+                TestState =
+                {
+                    OutputKind = OutputKind.ConsoleApplication
+                },
+                LanguageVersion = LanguageVersion.CSharp9
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task TestRedundantElseFix_VariableCollisionInIf()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode = """
+                    using System;
+                    
+                    class C
+                    {
+                        int M(int n) 
+                        {
+                            int i = 0;
+                            while (true)
+                            {
+                                if (i == n)
+                                {
+                                    i+=1;
+                                    int j = 0;
+                                    return j;
+                                }
+                                else
+                                {
+                                    int j = 0;
+                                    return j;
+                                }
+                            }
+                        }
+                    }
+                    """
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task TestRedundantElseFix_VariableCollisionInSeparateBlock()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode = """
+                    class C
+                    {
+                        int M(int n) 
+                        {
+                            int i = 0;
+                            while (true)
+                            {
+                                if (i < n)
+                                {
+                                    i+=1;
+                                    continue;
+                                }
+                                else
+                                {
+                                    int j = 0;
+                                }
+
+                    			{
+                    				int j = 0;
+                    			}
```

Comment:
why the odd formatting?

Reply:
ok.  you have tabs.  I recommend turniing on this "view whitespace" option:

![image](https://github.com/dotnet/roslyn/assets/4564579/0b739245-83e6-4ae6-8cd6-80e0ec603622)